core/src/components/banner/banner.tsx

```diff
@@ -115,10 +129,25 @@ export class TdsBanner {
           </div>
         )}
         <div class={`banner-content ${this.type} ${!this.icon ? 'no-icon' : ''}`}>
-          {this.header && <span class={`banner-header`}>{this.header}</span>}
-          {this.hasSubheader && <slot name="banner-subheader"></slot>}
+          {this.header ||
+            (this.hasHeaderSlot && (
+              <span class={`banner-header`}>
+                {this.header}
+                <slot name="banner-header"></slot>
+              </span>
+            ))}
+          {(this.hasSubheaderSlot || this.subheader) && (
+            <div class="banner-subheader">
+              {this.subheader}
+              <slot name="banner-subheader"></slot>
+            </div>
+          )}
```

Comment:
```suggestion
          {this.hasSubheaderSlot || (this.subheader && (
            <div class="banner-subheader">
              {this.subheader}
              <slot name="banner-subheader"></slot>
            </div>
          ))}
```

Or tweak this behavior instead?